discovery/gossiper.go

```diff
@@ -906,13 +906,24 @@ func (d *AuthenticatedGossiper) networkHandler() {
 			// have thousands of goroutines active.
 			validationBarrier.InitJobDependencies(announcement.msg)
 
+			d.wg.Add(1)
 			go func() {
+				defer d.wg.Done()
 				defer validationBarrier.CompleteJob()
 
 				// If this message has an existing dependency,
 				// then we'll wait until that has been fully
 				// validated before we proceed.
-				validationBarrier.WaitForDependants(announcement.msg)
+				err := validationBarrier.WaitForDependants(
+					announcement.msg,
+				)
+				if err == routing.ErrVBarrierShuttingDown {
```

Comment:
Know that atm this cannot happen, but should we also handle the case `err != nil`? I think something like this would make sense: 
```go
if err != nil {
    if err != ErrVBarrierShuttingDown {
         log.Error(err)
     }
     return
}
```

Reply:
Fixed!